File: 1838-frequency-of-the-most-frequent-element/1838-frequency-of-the-most-frequent-element.java
class Solution {
    public int maxFrequency(int[] nums, int k) {
        Arrays.sort(nums);
        int l = 0, r = 0;
        long total = 0, res = 0;
        while(r < nums.length){
            total += nums[r];
            while(r < nums.length && nums[r]*(r-l+1) > total + k){
                total -= nums[l++];
            }
            res = Math.max(res, r-l+1);
            r++;
        }
    return (int)res;
    }
}

File: 1838-frequency-of-the-most-frequent-element/1838-frequency-of-the-most-frequent-element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        l, r = 0, 0
        total, res = 0, 0
        nums.sort()
        while(r < len(nums)):
            total += nums[r]
            while r < len(nums) and nums[r] * (r-l+1) > total + k:
                total -= nums[l]
                l += 1
            res = max(res, r-l+1)
            r += 1
        return res